book/src/dev/rfcs/0005-state-updates.md

```diff
@@ -440,31 +440,44 @@ queued block (and any of its descendants) can be committed to the state
       - lookup the `block` for `hash`
       - remove `block` from `self.queued_blocks`
       - remove `hash` from `self.queued_by_height`
-      - let result = `self.commit_block(block)`;
-      - add `result` to `new_parents`
+      - let hash = `self.commit_block(block)`;
+      - add `hash` to `new_parents`
+
+### `fn commit_block(&mut self, block: QueuedBlock) -> block::Hash`
 
-### `fn commit_block(&mut self, block: QueuedBlock) -> Option<block::Hash>`
+Try to commit `block` to the non-finalized state. Must succeed, because
+`commit_block` is only called when `block` is ready to be committed.
 
-Try to commit `block` to the non-finalized state. Returns `None` if the block
-cannot be committed due to missing context.
+1. If the block is a pre-Sapling block, panic.
```

Comment:
For background, see https://github.com/ZcashFoundation/zebra/pull/902/files#r490654542

Reply:
> panic

![image](https://user-images.githubusercontent.com/552961/94095720-50ee6300-fdf0-11ea-9d98-95fa25224c93.png)